my-app/src/App.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import logo from './logo.svg'
+import './App.css';
+
+
+
+class GameBoard {
+  constructor( size ){
+    this.size = size
+    this.board = []
+    this.initializeBoard()
+  }
+
+  initializeBoard(){
+    for(let x = 0; x < this.size; x++){
+      this.board[x] = []
+      for (let y = 0; y < this.size; y++){
+        this.board[x][y] = new Gamepeice(x,y)
```

Comment:
Gamepeice gets used before it's been declared, and React issues a warning about this; you can take care of this by shifting your Gamepeice class up above GameBoard